fe/src/main/java/org/apache/doris/planner/SingleNodePlanner.java

```diff
@@ -327,6 +325,10 @@ private void turnOffPreAgg(AggregateInfo aggInfo, SelectStmt selectStmt, Analyze
                 break;
             }
 
+            if (((OlapScanNode)root).getForceOpenPreAgg()) {
+                return;//OlapScanNode has be forced opened PreAgg then the function return directly.
```

Comment:
why return?
you should call olapNode.setIsPreAggregation(true, null);

Reply:
if  return value of the function `getForceOpenPreAgg()` is true, the OlapScanNode has be set TRUE，so we don't call **`olapNode.setIsPreAggregation(true, null);`**